### PR TITLE
GCB issuance validation

### PR DESCRIPTION
**Pull request by ClemAtt**
## Description

Adds processes and validation for GCB issuance, inc:

- Add certificate global params to settings
- Certificate services and tests
- Device services and tests
- Adding meter_data_id attribute to device
- cqrs patches - see #61 
- Elexon client updates

Still to do:

Consider best way to batch issue certificates as started in function `issue_certificates_in_date_range`. Particularly:

- Whether to issue certificates for all devices on the registry at once in a loop?
- Whether to try and batch validate the certificates e.g. for bundle start and end values
- Which db session to use for different operations e.g. when querying for device capacity

## Has This Been Tested?

- [x] Yes - have added tests for the services, validation functions and issuance service

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: gc_registry/device/services.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    else:
+        return None
+
+
+def device_mw_capacity_to_wh_max(
+    device_capacity_mw: float, hours: float = settings.CERTIFICATE_GRANULARITY_HOURS
+) -> float:
+    """Take the device capacity in MW and calculate the maximum Watt-Hours
+    the device can produce in a given number of hours"""
+    WH_IN_MW = 1e6
```
Comment (Salamehar) [inline gc_registry/device/services.py:29]: `W_IN_MW `as it is 1000000 watts, not watts/hour, in a mega watt

**Review by Salamehar — COMMENTED** [review]
Minor suggestions

File: gc_registry/core/alembic/versions/2cb0ec2de122_gcb_device_issuance_normalisation.py [hunk 2]
```diff
@@ -63,6 +63,7 @@ def upgrade() -> None:
     op.add_column('granularcertificatebundle', sa.Column('metadata_id', sa.Integer(), nullable=False))
     op.add_column('granularcertificatebundle', sa.Column('is_storage', sa.Integer(), nullable=False))
     op.add_column('granularcertificatebundle', sa.Column('created_at', sa.DateTime(), nullable=False))
+    op.add_column('granularcertificatebundle', sa.Column('is_deleted', sa.Boolean(), nullable=False))
```
Comment (ClemAtt) [inline gc_registry/core/alembic/versions/2cb0ec2de122_gcb_device_issuance_normalisation.py:66]: This was missing from previous PR

**Review by Salamehar — APPROVED** [review]
LGTM 👍

**Pull request merged**